sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyLessECSignature.java

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.ProviderException;
+import java.security.PrivateKey;
+import java.security.SecureRandom;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.AlgorithmParameters;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.ECParameterSpec;
+import java.util.Base64;
+
+import static com.azure.security.keyvault.jca.KeyVaultClient.createKeyVaultClientBySystemProperty;
+import static com.azure.security.keyvault.jca.KeyVaultEncode.encodeByte;
+
+/**
+ * KeyVault EC signature to support key less
+ */
+public abstract class KeyVaultKeyLessECSignature extends AbstractKeyVaultKeyLessSignature {
+    // message digest implementation we use
+    private final MessageDigest messageDigest;
+
+    // signature parameters
+    private ECParameterSpec signatureParameters = null;
+
+    private final String keyVaultDigestName;
+
+    private KeyVaultClient keyVaultClient;
+
+    private String keyId;
+
+    /**
+     * Constructs a new KeyVaultKeyLessECSignature that will use the specified digest
+     */
+    KeyVaultKeyLessECSignature(String digestName, String keyVaultDigestName) {
+        setKeyVaultClient(createKeyVaultClientBySystemProperty());
+        try {
+            messageDigest = MessageDigest.getInstance(digestName);
+        } catch (NoSuchAlgorithmException e) {
+            throw new ProviderException(e);
+        }
+        this.keyVaultDigestName = keyVaultDigestName;
+    }
+
+    //add this for test
+    void setKeyVaultClient(KeyVaultClient keyVaultClient) {
+        this.keyVaultClient = keyVaultClient;
+    }
+
+    @Override
+    protected void engineInitSign(PrivateKey privateKey) {
+        engineInitSign(privateKey, null);
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as RSAPSSSignature#engineInitSign.
+    @Override
+    protected void engineInitSign(PrivateKey privateKey, SecureRandom random) {
```

Comment:
Can we move this method to `AbstractKeyVaultKeyLessSignature`? Same to some other methods in this class.